libs/langchain-anthropic/src/utils/message_inputs.ts

```diff
@@ -116,7 +116,8 @@ function _ensureMessageContents(
             content: [
               {
                 type: "tool_result",
-                content: _formatContent(message.content),
+                // rare case: message.content could be undefined
+                ...(message.content != null ? { content: _formatContent(message.content) } : {}),
```

Comment:
Slightly modified from the original submission. Anthropic's API reference shows `content` here as being an _optional_ (but not nullable) field of type `string | object[]`. The changes in `core` will make this case no longer possible, but this should handle it for older versions that weren't coalescing `undefined` tool output to `""`.